onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -279,7 +299,35 @@ static bool MatchPositionEmbeddingSubgraph1(
 
     subgraph_node_indices.push_back(shape_node_index);
   } else {  // gather_output_edges_count == 2
-    if (!MatchPositionSubgraph(graph, expand_node, input_ids, logger, subgraph_node_indices, gather_node.Index())) {
+    // Match optional Reshape -> Equal -> Where -> Expand
+    //                  |                  |
+    //                  --------------------
+    std::vector<const Node::EdgeEnd*> pg_edges_2;
+    std::vector<const Node::EdgeEnd*> pg_edges_3;
+    std::vector<graph_utils::EdgeEndToMatch> path_to_match_1{
+        {0, 1, "Where", {9}, kOnnxDomain},
+        {0, 0, "Equal", {1, 11}, kOnnxDomain},
+        {0, 0, "Reshape", {5}, kOnnxDomain}};
+    std::vector<graph_utils::EdgeEndToMatch> path_to_match_2{
+        {0, 1, "Where", {9}, kOnnxDomain},
+        {0, 2, "Reshape", {5}, kOnnxDomain}};
+    if (graph_utils::FindPath(expand_node, true, path_to_match_1, pg_edges_2, logger) &&
```

Comment:
Only need match path_to_match_1. After it is matched, you can verify the output node arg name of Reshape appears in input node arg names of Where.